{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "disease_signatures = pd.read_table(\"data/Disease_Signatures_MECFS.tsv\", sep=\"\\t\")\n",
    "heads = disease_signatures.columns\n",
    "disease_signatures[heads[1]] = disease_signatures[heads[1]].str.split(\", \")\n",
    "disease_signatures[heads[2]] = disease_signatures[heads[2]].str.split(\", \")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP ID</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Position in Build GRCh37</th>\n",
       "      <th>Variants (Ref/Alt)</th>\n",
       "      <th>Critical SNP</th>\n",
       "      <th>GWAS p-value (PLINK)</th>\n",
       "      <th>Gene(s) mapped based on genomic position</th>\n",
       "      <th>SNP to Gene mapping</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs41306603</td>\n",
       "      <td>1</td>\n",
       "      <td>33321877</td>\n",
       "      <td>A/C</td>\n",
       "      <td>1</td>\n",
       "      <td>0.542700</td>\n",
       "      <td>S100PBP</td>\n",
       "      <td>genebody</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs2985754</td>\n",
       "      <td>1</td>\n",
       "      <td>39053684</td>\n",
       "      <td>T/C</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000646</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs946448</td>\n",
       "      <td>1</td>\n",
       "      <td>54661150</td>\n",
       "      <td>C/T</td>\n",
       "      <td>0</td>\n",
       "      <td>0.041750</td>\n",
       "      <td>CYB5RL, AL357673.1, MRPL37</td>\n",
       "      <td>genebody</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs12562306</td>\n",
       "      <td>1</td>\n",
       "      <td>54662686</td>\n",
       "      <td>G/A</td>\n",
       "      <td>0</td>\n",
       "      <td>0.039590</td>\n",
       "      <td>CYB5RL, AL357673.1, MRPL37</td>\n",
       "      <td>genebody</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs17571877</td>\n",
       "      <td>1</td>\n",
       "      <td>73987668</td>\n",
       "      <td>G/A</td>\n",
       "      <td>0</td>\n",
       "      <td>0.368700</td>\n",
       "      <td>LINC02238</td>\n",
       "      <td>closest</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>194</th>\n",
       "      <td>rs10470164</td>\n",
       "      <td>21</td>\n",
       "      <td>33384319</td>\n",
       "      <td>T/C</td>\n",
       "      <td>0</td>\n",
       "      <td>0.615600</td>\n",
       "      <td>HUNK</td>\n",
       "      <td>genebody</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>195</th>\n",
       "      <td>rs13047599</td>\n",
       "      <td>21</td>\n",
       "      <td>34926260</td>\n",
       "      <td>T/C</td>\n",
       "      <td>0</td>\n",
       "      <td>0.159700</td>\n",
       "      <td>SON</td>\n",
       "      <td>genebody</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>196</th>\n",
       "      <td>rs2309099</td>\n",
       "      <td>21</td>\n",
       "      <td>43074513</td>\n",
       "      <td>T/G</td>\n",
       "      <td>0</td>\n",
       "      <td>0.004779</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197</th>\n",
       "      <td>rs11910362</td>\n",
       "      <td>21</td>\n",
       "      <td>44042101</td>\n",
       "      <td>T/C</td>\n",
       "      <td>0</td>\n",
       "      <td>0.041830</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>198</th>\n",
       "      <td>rs5747306</td>\n",
       "      <td>22</td>\n",
       "      <td>18128960</td>\n",
       "      <td>A/G</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000648</td>\n",
       "      <td>BCL2L13</td>\n",
       "      <td>genebody</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>199 rows Ã— 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         SNP ID  Chromosome  Position in Build GRCh37 Variants (Ref/Alt)  \\\n",
       "0    rs41306603           1                  33321877                A/C   \n",
       "1     rs2985754           1                  39053684                T/C   \n",
       "2      rs946448           1                  54661150                C/T   \n",
       "3    rs12562306           1                  54662686                G/A   \n",
       "4    rs17571877           1                  73987668                G/A   \n",
       "..          ...         ...                       ...                ...   \n",
       "194  rs10470164          21                  33384319                T/C   \n",
       "195  rs13047599          21                  34926260                T/C   \n",
       "196   rs2309099          21                  43074513                T/G   \n",
       "197  rs11910362          21                  44042101                T/C   \n",
       "198   rs5747306          22                  18128960                A/G   \n",
       "\n",
       "     Critical SNP  GWAS p-value (PLINK)  \\\n",
       "0               1              0.542700   \n",
       "1               0              0.000646   \n",
       "2               0              0.041750   \n",
       "3               0              0.039590   \n",
       "4               0              0.368700   \n",
       "..            ...                   ...   \n",
       "194             0              0.615600   \n",
       "195             0              0.159700   \n",
       "196             0              0.004779   \n",
       "197             0              0.041830   \n",
       "198             0              0.000648   \n",
       "\n",
       "    Gene(s) mapped based on genomic position SNP to Gene mapping  \n",
       "0                                    S100PBP            genebody  \n",
       "1                                        NaN                 NaN  \n",
       "2                 CYB5RL, AL357673.1, MRPL37            genebody  \n",
       "3                 CYB5RL, AL357673.1, MRPL37            genebody  \n",
       "4                                  LINC02238             closest  \n",
       "..                                       ...                 ...  \n",
       "194                                     HUNK            genebody  \n",
       "195                                      SON            genebody  \n",
       "196                                      NaN                 NaN  \n",
       "197                                      NaN                 NaN  \n",
       "198                                  BCL2L13            genebody  \n",
       "\n",
       "[199 rows x 8 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "snp_table = pd.read_table(\"data/SNP_Table_MECFS.tsv\", sep=\"\\t\")\n",
    "snp_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SUPPLEMENTARY FILE 2 - METABOLON CLP DATA</th>\n",
       "      <th>Unnamed: 1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SHEET</td>\n",
       "      <td>CONTENT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1.Suppl 2 content</td>\n",
       "      <td>Overview of the content in Supplermentary file 2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2.FC &amp; Stats</td>\n",
       "      <td>Curated CLP metabolite data ME-ALL-M1-M2-M3 vs HC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3.FC CLP classes</td>\n",
       "      <td>Total lipid class concentrations (calculated/e...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>4.psign MEall</td>\n",
       "      <td>p-significant metabolites, MEall vs HC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>5.psign Uniform</td>\n",
       "      <td>p-significant metabolites, Uniform in ME-M1 an...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>6.psign ME-M1</td>\n",
       "      <td>p-significant metabolites, ME-M1 vs HC, UP &amp; D...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>7.psign ME-M2</td>\n",
       "      <td>p-significant metabolites, ME-M2 vs HC, UP &amp; D...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>8.psign ME-M3</td>\n",
       "      <td>p-significant metabolites, ME-M3 vs HC, UP &amp; D...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>9.FA Conc</td>\n",
       "      <td>Total fatty acid concentration, per class, cal...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>10.FA Mol%</td>\n",
       "      <td>Fatty acid composition, per class, calculated ...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   SUPPLEMENTARY FILE 2 - METABOLON CLP DATA  \\\n",
       "0                                        NaN   \n",
       "1                                      SHEET   \n",
       "2                         1.Suppl 2 content    \n",
       "3                               2.FC & Stats   \n",
       "4                           3.FC CLP classes   \n",
       "5                              4.psign MEall   \n",
       "6                            5.psign Uniform   \n",
       "7                              6.psign ME-M1   \n",
       "8                              7.psign ME-M2   \n",
       "9                              8.psign ME-M3   \n",
       "10                                 9.FA Conc   \n",
       "11                                10.FA Mol%   \n",
       "\n",
       "                                           Unnamed: 1  \n",
       "0                                                 NaN  \n",
       "1                                             CONTENT  \n",
       "2   Overview of the content in Supplermentary file 2   \n",
       "3   Curated CLP metabolite data ME-ALL-M1-M2-M3 vs HC  \n",
       "4   Total lipid class concentrations (calculated/e...  \n",
       "5             p-significant metabolites, MEall vs HC   \n",
       "6   p-significant metabolites, Uniform in ME-M1 an...  \n",
       "7   p-significant metabolites, ME-M1 vs HC, UP & D...  \n",
       "8   p-significant metabolites, ME-M2 vs HC, UP & D...  \n",
       "9   p-significant metabolites, ME-M3 vs HC, UP & D...  \n",
       "10  Total fatty acid concentration, per class, cal...  \n",
       "11  Fatty acid composition, per class, calculated ...  "
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metabolites = pd.read_excel(\"data/Hoel_et_al_metabolites.xlsx\")\n",
    "metabolites"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SUPPLEMENTARY FILE 2 - METABOLON CLP DATA</th>\n",
       "      <th>Unnamed: 1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SHEET</td>\n",
       "      <td>CONTENT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1.Suppl 2 content</td>\n",
       "      <td>Overview of the content in Supplermentary file 2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2.FC &amp; Stats</td>\n",
       "      <td>Curated CLP metabolite data ME-ALL-M1-M2-M3 vs HC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3.FC CLP classes</td>\n",
       "      <td>Total lipid class concentrations (calculated/e...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  SUPPLEMENTARY FILE 2 - METABOLON CLP DATA  \\\n",
       "0                                       NaN   \n",
       "1                                     SHEET   \n",
       "2                        1.Suppl 2 content    \n",
       "3                              2.FC & Stats   \n",
       "4                          3.FC CLP classes   \n",
       "\n",
       "                                          Unnamed: 1  \n",
       "0                                                NaN  \n",
       "1                                            CONTENT  \n",
       "2  Overview of the content in Supplermentary file 2   \n",
       "3  Curated CLP metabolite data ME-ALL-M1-M2-M3 vs HC  \n",
       "4  Total lipid class concentrations (calculated/e...  "
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metabolites.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
